sound/soc/sof/intel/byt.c

```diff
@@ -826,11 +842,19 @@ const struct snd_sof_dsp_ops sof_cht_ops = {
 	.send_msg	= byt_send_msg,
 	.fw_ready	= byt_fw_ready,
 
+	.ipc_read	= intel_ipc_read,
+	.ipc_read_pcm_stream = intel_ipc_read_pcm_stream,
+	.ipc_read_pcm_params = intel_ipc_read_pcm_params,
+
 	/* debug */
 	.debug_map	= cht_debugfs,
 	.debug_map_count	= ARRAY_SIZE(cht_debugfs),
 	.dbg_dump	= byt_dump,
 
+	/* stream callbacks */
+	.pcm_open	= intel_pcm_open,
+	.pcm_close	= intel_pcm_close,
+
```

Comment:
I am a bit lost here. Why are pcm_open and pcm_close part of an IPC refactoring, and they are only used for legacy platforms?

Reply:
because HDA-platforms already have .pcm_open() and .pcm_close(), which point to hda_dsp_pcm_open() and hda_dsp_pcm_close() respectively. Now with this refactoring non-HDA platforms also need those, that was the only way I found to get to per-stream position offset from their .ipc_read_pcm_params() and .ipc_read_pcm_stream()